src/_P088_HeatpumpIR.ino

```diff
@@ -295,4 +283,18 @@ boolean Plugin_088(byte function, struct EventStruct *event, String& string)
   return success;
 }
 
+#ifndef PLUGIN_035
+void enableIR_RX(boolean enable)
+{
+#ifdef PLUGIN_016
+  if (irReceiver == 0) return;
+  if (enable) {
+    irReceiver->enableIRIn(); // Start the receiver
+  } else {
+    irReceiver->disableIRIn(); // Stop the receiver
+  }
+#endif //PLUGIN_016
+}
+#endif //PLUGIN_035
```

Comment:
This will become rather complicated in the end.
P035 must NOT be available and P016 MUST be available for this?
What if P035 IS available? Then this will not compile I guess?
Or is this function also available in P035?

I think specific functions which depend on other plugins, should be moved to some IR_plugin_helper.ino file.
Then the specifics are dealt with in a single file which makes it much easier (or less difficult) to manage all special cases.

Reply:
this function also available in P035 and it is very unlikly that it will change ever.
I can move it to P016 and make it available from there I guess...

Reply:
How about adding in the end of P016. So this function will apear once and in the plugin that changes the state to.

```
......................
#endif // USES_P016

#if defined(PLUGIN_035) || defined(PLUGIN_088)
void enableIR_RX(boolean enable)
{
#ifdef PLUGIN_016
  if (irReceiver == 0) return;
  if (enable) {
    irReceiver->enableIRIn(); // Start the receiver
  } else {
    irReceiver->disableIRIn(); // Stop the receiver
  }
#endif //PLUGIN_016
}
#endif //PLUGIN_035 || PLUGIN_088
```

Reply:
No, what I meant was to create a separate file where the inter-plugin dependencies are being dealt with.
Now we have code in P016 depending on P035 and vice verse.
2 of these are kinda manageable, but with nr 3 (P088) it will become a mess.
So it is best to have this split function (which I like) moved to a separate file and also try to move other inter-plugin dependencies related to IR to a separate file.
Then you don't have the #ifdef <other plugin> kind of stuff in the plugin files and the only thing they have to refer to is the functions in this extra file.
So when you're looking at the code in this extra file, you will immediately see where it is used, as it must have wrappers like `#if defined(USES_P035) || defined(USES_P088)`.

Having duplicate functions in several files only made "unique" by using #ifdef statements is prone to errors.
When someone is making a new plugin P099 which does need one of those functions, then it depends on the search results of that user which one he might pick. So that's going to end in a messy situation.

Reply:
Cool then 

```
......................
#endif // USES_P016

void enableIR_RX(boolean enable)
{
#ifdef PLUGIN_016
  if (irReceiver == 0) return;
  if (enable) {
    irReceiver->enableIRIn(); // Start the receiver
  } else {
    irReceiver->disableIRIn(); // Stop the receiver
  }
#endif //PLUGIN_016
}
```
in the end of P016 will suffice. If I have more functions that will benefit from moving them to a seperate 'helper' file I will make one

Reply:
Looks good to me.
Nice intermediate solution, where you don't have defines of other plugins in that file.